gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -435,10 +452,22 @@ public void run() {
       try {
         GobblinServiceJobScheduler.this.runJob(this.jobConfig, this.jobListener);
         if (flowCatalog.isPresent() && removeSpec) {
+          if (GobblinServiceJobScheduler.this.flowCatalog.get().getSpecSyncObjects().containsKey(specUri.toString())) {
+            // if the sync object does not exist, this job must be set to run due to job submission at service restart
+            Object syncObject = GobblinServiceJobScheduler.this.flowCatalog.get().getSpecSyncObjects().get(specUri.toString());
```

Comment:
This looks very cumbersome! Let's add a method in flowCatalog that returns the syncObject given the specUri and avoid exposing specSyncObjects map outside flowcatalog.